### DISCUSSION TITLE
ClassifyConversion should also include conversion to ITypeParameterSymbol

### DISCUSSION THREAD

**Discussion by B1Z0N** [dotnet/roslyn#74552] [Ideas] [answered]
I think this will be more logical if we could classify conversion to a generic with set of constraints as possible, because it's possible. I think it's a fitting point to insert this functionality. Although, checking if a set of constraints denoted by generic is satisfied with a said type as a separate method somewhere in the API would be useful because this functionality is simply missing as was noted 4 years ago here #46998.

It seems like language functionality is there, but Roslyn API can't keep up and cover it properly. Is this a desired change from the perspective of Roslyn devs? The community voiced the need for this. I'd like to make it work.

**Comment by 333fred** [comment, marked as answer]
It certainly couldn't be `ClassifyConversion`; attempting to convert from some concrete type to an `ITypeParameterSymbol` already has meaning. It would have to be a new API, and go through our [API Review Process](https://github.com/dotnet/roslyn/blob/main/docs/contributing/API%20Review%20Process.md). As yet, no one has gone through this process; if you wanted to go through the process, we'd be happy to review your proposal.

**Reply by B1Z0N** [reply]
Will do
